Makefile

```
@@ -60,6 +60,7 @@ setup:
 	docker-compose up -d
 	docker-compose exec -T php composer install
 	scripts/reload-local.sh --site=${DEFAULT_SITE_ALIAS}
+	docker-compose run -e'PHP_ERROR_REPORTING=E_ALL & ~E_DEPRECATED' --rm -T php 'vendor/bin/grumphp' 'git:init'
```

I have one question, why is it a problem that the warnings appear? Does the warnings produce a problem to run the git:init command?

If there are no problems regenerating the git hooks I would prefer to show the warnings. Warnings must be fixed, but they may be ignored instead if they are hidden.

---

I also recommend using docker-compose exec instead of run. In this step the PHP container is already up, so there is no need to create a extra container with docker-compose run.

Here are the two changes I propose:

```suggestion
	docker-compose exec -T php 'vendor/bin/grumphp' 'git:init'
```

---

I used run to allow the command to be moved outside the make setup to something like make setup_grumphp, but this could be done in the future. For now it's ok to use the exec variation, thank you!

---

The deprecation warnings in this case cannot be fixed. They come from GrumPHP which in the current version is not fully compatible with PHP 8. These warnings should disappear when we can update grumphp, but in the meantime we have no choice but to ignore them because they mess the terminal output.

In fact, grumphp if configured correctly still warns if it detects deprecated code in the commit, so I don't see a problem hiding these warnings.

---

About using run or exec, in this case we have to use run because the environment variable is used to configure the php.ini and it is not taken into account once the container is launched, so if we use docker-compose exec -e for the error_reporting the warnings are still seen.

---

We have talked the last time that the environment variables can be added in docker-compose exec.

Example:

```bash
docker-compose exec -T -e 'PHP_ERROR_REPORTING=E_ALL & ~E_DEPRECATED'  php 'vendor/bin/grumphp' 'git:init'
```

If you haven't tried it yet to do then I will.

---

My bad. I didn't read well your message and the docker-compose exec solution does not work. Then it is okay to user docker-compose run.